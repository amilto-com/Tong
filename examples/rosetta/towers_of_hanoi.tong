// Rosetta Code: Towers of Hanoi
fn hanoi(n, from, to, aux) {
    if n == 0 {
        // do nothing
    } else {
        hanoi(n-1, from, aux, to)
        print("Move", from, "->", to)
        hanoi(n-1, aux, to, from)
    }
}

hanoi(4, "A", "C", "B")
