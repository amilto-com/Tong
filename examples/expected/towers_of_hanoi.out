Move A -> B Move A -> B Move A -> C Move B -> C Move B -> C Move A -> B Move A -> B Move A -> C Move B -> C Move B -> C Move A -> B Move C -> A Move C -> A Move C -> B Move A -> B Move A -> B Move C -> A Move C -> A Move C -> B Move A -> B Move A -> B Move A -> B Move A -> B Move A -> C Move B -> C Move B -> C Move A -> B Move A -> B Move A -> C Move B -> C Move B -> C Move A -> B Move C -> A Move C -> A Move C -> B Move A -> B Move A -> B Move C -> A Move C -> A Move C -> B Move A -> B Move A -> B Move A -> C Move B -> C Move B -> C Move B -> A Move C -> A Move C -> A Move B -> C Move B -> C Move B -> A Move C -> A Move C -> A Move B -> C Move A -> B Move A -> B Move A -> C Move B -> C Move B -> C Move A -> B Move A -> B Move A -> C Move B -> C Move B -> C Move B -> C Move B -> C Move B -> A Move C -> A Move C -> A Move B -> C Move B -> C Move B -> A Move C -> A Move C -> A Move B -> C Move A -> B Move A -> B Move A -> C Move B -> C Move B -> C Move A -> B Move A -> B Move A -> C Move B -> C Move B -> C